### ISSUE TITLE
except NameError: error: "var" is possibly unbound

### ISSUE THREAD

**Issue by sebastian-philipp** [microsoft/pyright#1363]
**Describe the bug**

pyright complains about a variable being unbound, but the variable is actually bound. 

**Expected behavior**

no warning

**Screenshots or Code**

```python
if False:
    var = 1

try:
    var  # type: ignore
except NameError:
    var = 1

print(var)  # pyright:   9:7 - error: "var" is possibly unbound (reportUnboundVariable)
```

**VS Code extension or command-line**

```
$ pyright --version
pyright 1.1.101
$ pyright file.py
No configuration file found.
stubPath /home/sebastian/Repos/ceph/src/cephadm/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
/home/sebastian/Repos/ceph/src/cephadm/file.py
  9:7 - error: "var" is possibly unbound (reportUnboundVariable)
1 error, 0 warnings, 0 infos 
Completed in 0.629sec
```

Are you running pyright as a VS Code extension or a command-line tool? Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.

**Comment by erictraut** [comment]
It would be possible to address this particular simple example by adding extra logic to handle this, but using a `NameError` exception handler to address unbound variables is not a common or recommended pattern. Try/except statements were designed for exceptions, not for standard code flow. I suggest you should change your code. Alternatively, you can suppress the error using a `# type: ignore` statement on the line where `var` is used. Or if you use this pattern frequently in your code, you can disable the `reportUnboundVariable` check.

**Comment by sebastian-philipp** [comment]
Actually this is a common idiom for dealing with (standard) library changes:

```
try:
    TimeoutError  # >= Python 3.3
except NameError:
    TimeoutError = OSError  # < Python 3.3
```
